Python/ceval_macros.h

```diff
@@ -296,11 +296,19 @@ GETITEM(PyObject *v, Py_ssize_t i) {
 #define ADAPTIVE_COUNTER_IS_MAX(COUNTER) \
     (((COUNTER) >> ADAPTIVE_BACKOFF_BITS) == ((1 << MAX_BACKOFF_VALUE) - 1))
 
+#ifdef Py_GIL_DISABLED
+#define DECREMENT_ADAPTIVE_COUNTER(COUNTER)                           \
```

Comment:
I would instead do something that causes a compile-time error if this macro is even expanded -- it should only occur inside `#if ENABLE_SPECIALIZATION`.